### DISCUSSION TITLE
The T.RandomLinkSplit seems not work with Non Plaintoid Dataset

### DISCUSSION THREAD

**Discussion by HanChen-HUST** [pyg-team/pytorch_geometric#5709] [Q&A]
I am doing link prediction task,and need to divide the dataset,code like this

transform = T.Compose([
        T.NormalizeFeatures(),
        T.ToDevice(device),
        T.RandomLinkSplit(num_val=0., num_test=0.1, is_undirected=True,
                          add_negative_train_samples=False),
    ])
dataset=Amazon(root=path, name='photo', transform=transform)
it works with Plaintoid dataset like Cora,but it wouldnt work with Non Plaintoid dataset,such ad Amazon photo
but it occurs a error "Insufficient number of edges for training." how can I solve it,thanks

**Comment by EdisonLeeeee** [comment]
Which PyG version are you using? It looks like this bug was resolved in 2.1.0.

**Reply by HanChen-HUST** [reply]
Hi,the pyg version I used is 2.0.1

**Reply by EdisonLeeeee** [reply]
Can you upgrade your PyG to the latest version and try it again? Let me know if the error still exists.
